app/bundles/DashboardBundle/Config/config.php

```diff
@@ -42,13 +42,22 @@
     'services' => [
         'other' => [
             'mautic.dashboard.widget' => [
-                'class'     => \Mautic\DashboardBundle\Dashboard\Widget::class,
+                'class'     => Mautic\DashboardBundle\Dashboard\Widget::class,
                 'arguments' => [
                     'mautic.dashboard.model.dashboard',
                     'mautic.helper.user',
                     'session',
                 ],
             ],
+            'mautic.dashboard.widget_detail.event.factory' => [
+                'class'     => Mautic\DashboardBundle\Event\WidgetDetailEventFactory::class,
+                'arguments' => [
+                    'translator',
+                    'mautic.cache.provider',
+                    'mautic.security',
+                    'mautic.helper.user',
+                ],
+            ],
```

Comment:
This can be removed as M5 comes with autowiring
```suggestion
```

Reply:
By default we are ignoring autowiring for Event directory so I had added this as service but now I have loaded This factory in autowiring service file.

Reply:
@dadarya0 I see. In that case I'd recommend to move the `WidgetDetailEventFactory` to another folder because we shouldn't be mixing service classes with DTO classes. Can you move it to another folder? Like `app/bundles/DashboardBundle/Factory/WidgetDetailEventFactory.php`?

Reply:
Moved to new folder named Factory